app/sagas/init.js

```diff
@@ -14,22 +14,18 @@ import database from '../lib/realm';
 
 const restore = function* restore() {
 	try {
-		const { token, server } = yield all({
-			token: AsyncStorage.getItem(RocketChat.TOKEN_KEY),
-			server: AsyncStorage.getItem('currentServer')
-		});
-
+		const { serversDB } = database.databases;
+		const currentServer = yield serversDB.objects('servers').filtered('currentServer = true');
+		const { id: server = null } = currentServer.length && currentServer[0];
+		const { password: token } = server ? yield Keychain.getInternetCredentials(server) : { password: null };
```

Comment:
const { password: token = null } = server && yield Keychain.getInternetCredentials(server);

Reply:
nice man!

Reply:
@IlarionHalushka This doesn't works :( because if server fail will return undefined and try to access password on undefined.